### DISCUSSION TITLE
problem with set specific configuration in new version 3.0.0

### DISCUSSION THREAD

**Discussion by meysamhadeli** [testcontainers/testcontainers-dotnet#830] [Q&A] [answered]
Hi. 
I have a problem with configuration containers in new version.
I created an example for adding my custom configuration, but always get default value that set in the base! For example, I always get this value `username: rabbitmq`, `password:`, `image: rabbitmq:3.11`. But in my code I set a new value for each one.
Here is my custom configuration for RabbitMQ in the code:
```csharp
    public static RabbitMqContainer RabbitMqTestContainer()
    {
        var builder = new RabbitMqBuilder();

        builder.WithImage("rabbitmq:3-management");
        builder.WithUsername("guest");
        builder.WithPassword("guest");

        return builder.Build();
    }
```
Did I miss something?

**Comment by HofmeisterAn** [comment, marked as answer]
> I have a problem with configuration containers in new version.

The builder has always been immutable (data held inside it). It always creates a new instance. Please see my answer here: https://github.com/testcontainers/testcontainers-dotnet/discussions/713#discussioncomment-4397966.

**Reply by meysamhadeli** [reply]
Thanks. It was helpful :)
